### PR TITLE
supporting new variables

### PR DESCRIPTION
**Pull request by Devinwong**
feat:  supporting new variables

-->

Fixes # 

**Description**
 supporting new variables
**How was this change tested?**
Tested with manually scale up nodes
*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl [hunk 1]
```diff
@@ -65,23 +65,23 @@ TELEPORT_ENABLED="{{getBoolFromFeatureState .TeleportConfig.Status}}"
 TELEPORTD_PLUGIN_DOWNLOAD_URL={{.TeleportConfig.TeleportdPluginDownloadUrl}}
 RUNC_VERSION={{.RuncConfig.RuncVersion}}
 RUNC_PACKAGE_URL={{.RuncConfig.RuncPackageUrl}}
-ENABLE_HOSTS_CONFIG_AGENT="{{getBoolFromFeatureState .HostsConfigAgentStatus}}"
-DISABLE_SSH={{if .SshStatus}}"false"{{else}}"true"{{end}}
+ENABLE_HOSTS_CONFIG_AGENT="{{.EnableHostsConfigAgent}}"
+DISABLE_SSH="{{not .EnableSsh}}"
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: if EnableSsh is not specified, the default value will be false which means DISABLE_SSH will be true correct?  Is this what we currently do?

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl [hunk 2]
```diff
@@ -65,23 +65,23 @@ TELEPORT_ENABLED="{{getBoolFromFeatureState .TeleportConfig.Status}}"
...
 CUSTOM_CA_CERT_{{$i}}="{{$cert}}"
 {{end}}
-IS_KRUSTLET="{{.IsKrustlet}}"
-IPV6_DUAL_STACK_ENABLED={{getBoolStringFromFeatureStatePtr .Ipv6DualStackEnabled}}
-ENABLE_UNATTENDED_UPGRADES={{getBoolStringFromFeatureStatePtr .UnattendedUpgradeStatus}}
-ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE={{derefBool .EnsureNoDupePromiscuousBridge}}
+IS_KRUSTLET="{{getIsKrustlet .WorkloadRuntime}}"
+IPV6_DUAL_STACK_ENABLED="{{.Ipv6DualStackEnabled}}"
+ENABLE_UNATTENDED_UPGRADES={{.EnableUnattendedUpgrade}}
+ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE={{getEnsureNoDupePromiscuousBridge .NetworkConfig.NetworkPlugin .NetworkConfig.NetworkPolicy}}
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: can you use the getters here? I tested and you should be able to call them from the template, e.g. `.NetworkConfig.GetNetworkPlugin`

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl [hunk 3]
```diff
@@ -101,6 +101,7 @@ KUBELET_CONFIG_FILE_CONTENT="{{.KubeletConfig.KubeletConfigFileContent}}"
 CUSTOM_SEARCH_DOMAIN_NAME="{{.CustomSearchDomain.CustomSearchDomainName}}"
 CUSTOM_SEARCH_REALM_USER="{{.CustomSearchDomain.CustomSearchDomainRealmUser}}"
 CUSTOM_SEARCH_REALM_PASSWORD="{{.CustomSearchDomain.CustomSearchDomainRealmPassword}}"
+HAS_CUSTOM_SEARCH_DOMAIN="{{getHasSearchDomain .CustomSearchDomain.CustomSearchDomainName .CustomSearchDomain.CustomSearchDomainRealmUser .CustomSearchDomain.CustomSearchDomainRealmPassword}}"
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: same as above, can you getters

Comment (lilypan26) [reply pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: also would it be better to just pass in `CustomSearchDomain` to the function? and then use the getters in the function?

File: pkg/providers/imagefamily/bootstrap/parser_helpers.go [hunk 4]
```diff
@@ -219,3 +230,21 @@ func getTLSBootstrapToken(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig)
 func getCustomSecureTLSBootstrapAADServerAppID(bootstrapConfig *nbcontractv1.TLSBootstrappingConfig) string {
 	return bootstrapConfig.GetCustomSecureTlsBootstrapAppserverAppid()
 }
+
+func getIsKrustlet(wr nbcontractv1.WorkloadRuntime) bool {
+	if wr == nbcontractv1.WorkloadRuntime_WASM_WASI {
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/parser_helpers.go]: nit: can simplify this to `return wr == nbcontractv1.WorkloadRuntime_WASM_WASI`

Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/parser_helpers.go]: Updated. Thanks

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl (hunk 3 above)
Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: Good suggestions. And I also changed all of them to getters. I forgot about this.

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl (hunk 2 above)
Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: Changed all of them to getters.

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl (hunk 1 above)
Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: Correct. We didn't add the optional label to `EnableSsh` so the default value of it is `false` if it's unspecified. But I changed it to getter too for consistency. Thanks.

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl [hunk 5]
```diff
@@ -65,23 +65,23 @@ TELEPORT_ENABLED="{{getBoolFromFeatureState .TeleportConfig.Status}}"
...
 CUSTOM_CA_CERT_{{$i}}="{{$cert}}"
 {{end}}
-IS_KRUSTLET="{{.IsKrustlet}}"
-IPV6_DUAL_STACK_ENABLED={{getBoolStringFromFeatureStatePtr .Ipv6DualStackEnabled}}
-ENABLE_UNATTENDED_UPGRADES={{getBoolStringFromFeatureStatePtr .UnattendedUpgradeStatus}}
-ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE={{derefBool .EnsureNoDupePromiscuousBridge}}
+IS_KRUSTLET="{{getIsKrustlet .GetWorkloadRuntime}}"
+IPV6_DUAL_STACK_ENABLED="{{.GetIpv6DualStackEnabled}}"
+ENABLE_UNATTENDED_UPGRADES={{.GetEnableUnattendedUpgrade}}
+ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE={{getEnsureNoDupePromiscuousBridge .GetNetworkConfig.GetNetworkPlugin .GetNetworkConfig.GetNetworkPolicy}}
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: nit, but could also just pass in the network config and then access the fields in the function. Can be done later though, approving.

**Review by lilypan26 — APPROVED** [review]

File: pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl (hunk 5 above)
Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl]: I was thinking if for our go binary if it's easier for it pass the individual variable or the whole network config object. But I think the whole object is easier. I will follow your suggestion. Thanks :)

**Pull request merged**